packages/libs/src/sdk/services/DiscoveryNodeSelector/DiscoveryNodeSelector.ts

```diff
@@ -267,7 +267,12 @@ export class DiscoveryNodeSelector implements DiscoveryNodeSelectorService {
   /**
    * Gets the list of services
    */
-  public getServices() {
+  public async getServices() {
+    const selected = await this.getSelectedEndpoint()
```

Comment:
why? this seems like a hidden side effect

Reply:
I don't love it, but without it, we don't update the services list - i figured it makes more sense to expect `getServices()` to be async and actually fetch the current service list than to sometimes return the bootstrap list if you haven't selected yet. This ensures the service list is always "fresh" at least from the selected DNs perspective